core/src/main/java/de/jplag/strategy/AbstractComparisonStrategy.java

```diff
@@ -45,8 +45,16 @@ protected void compareSubmissionsToBaseCode(SubmissionSet submissionSet) {
      * Compares two submissions and optionally returns the results if similarity is high enough.
      */
     protected Optional<JPlagComparison> compareSubmissions(Submission first, Submission second) {
+        return compareSubmissions(new SubmissionTuple(first, second));
+    }
+
+    protected Optional<JPlagComparison> compareSubmissions(SubmissionTuple tuple) {
+        Submission first = tuple.left();
+        Submission second = tuple.right();
+
         JPlagComparison comparison = greedyStringTiling.compare(first, second);
         logger.info("Comparing {}-{}: {}", first.getName(), second.getName(), comparison.similarity());
+        this.options.compareHook().callback(tuple);
```

Comment:
You could use the [BiConsumer](https://docs.oracle.com/javase/8/docs/api/java/util/function/BiConsumer.html) interface to reduce refactorings :)